go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1953,32 +1957,32 @@ func (tsv *TabletServer) QueryPlanCacheWait() {
 
 // SetMaxResultSize changes the max result size to the specified value.
 func (tsv *TabletServer) SetMaxResultSize(val int) {
-	tsv.qe.maxResultSize.Set(int64(val))
+	tsv.qe.maxResultSize.Store(int64(val))
 }
 
 // MaxResultSize returns the max result size.
 func (tsv *TabletServer) MaxResultSize() int {
-	return int(tsv.qe.maxResultSize.Get())
+	return int(tsv.qe.maxResultSize.Load())
 }
 
 // SetWarnResultSize changes the warn result size to the specified value.
 func (tsv *TabletServer) SetWarnResultSize(val int) {
-	tsv.qe.warnResultSize.Set(int64(val))
+	tsv.qe.warnResultSize.Store(int64(val))
 }
 
 // WarnResultSize returns the warn result size.
 func (tsv *TabletServer) WarnResultSize() int {
-	return int(tsv.qe.warnResultSize.Get())
+	return int(tsv.qe.warnResultSize.Load())
 }
 
 // SetThrottleMetricThreshold changes the throttler metric threshold
 func (tsv *TabletServer) SetThrottleMetricThreshold(val float64) {
-	tsv.lagThrottler.MetricsThreshold.Set(val)
+	tsv.lagThrottler.StoreMetricsThreshold(val)
 }
 
 // ThrottleMetricThreshold returns the throttler metric threshold
 func (tsv *TabletServer) ThrottleMetricThreshold() float64 {
-	return tsv.lagThrottler.MetricsThreshold.Get()
+	return time.Duration(tsv.lagThrottler.MetricsThreshold.Load()).Seconds()
```

Comment:
This one is incorrect. By default, the metrics threshold indicates seconds of replication lag. But it is also customizable to indicate any floating point value, per `metrics_query`. So, for example, it could be the number of threads running (an integer threshold), or the number of writes per sec (float).

Reply:
The correct behavior is to just return the value, without casting-to-Duration-and-extracting-Seconds()

Reply:
@shlomi-noach Ah ok, it does have to be a float and there's no float atomics (see https://github.com/golang/go/issues/21996), so now it uses the bit conversions. That's also what `sync2` was already doing anyway.

Reply:
right! Sadly we need to do the conversion. Floats in computers are a pain.

Reply:
This hasn't been addressed yet!

Reply:
Doh, fixed all cases except this one 🤦. Done!